### PR TITLE
fix outer loop

### PR DESCRIPTION
**Pull request by WalterBright**
Should check vtblFinal[ ] and then baseClass.

### CODE REVIEW

**Comment by MartinNowak** [comment]
Test case, bug report, description of fix?
About as useful as the comment on vtblFinals "More FuncDeclaration's that aren't in vtbl[]".

Looks sensible but it's not possible to assess the impact of the fix.

File: src/dclass.d [hunk 1]
```diff
@@ -1313,10 +1313,10 @@ public:
...
                 //else printf("\t\t%d\n", fd->type->covariant(tf));
             }
-            if (!cd)
-                break;
-            vtbl = &cd.vtblFinal;
-            cd = cd.baseClass;
+            if (vtbl != &cd.vtblFinal)
+                vtbl = &cd.vtblFinal;
+            else
+                cd = cd.baseClass;
```
Comment (MartinNowak) [inline src/dclass.d]: Dealing with vtbl->vtblFinal in the first loop iteration is very confusing. Move the loop body into a nested function instead and call
```d
searchVtbl(cd.vtbl);
while (cd)
{
    searchVtbl(cd.vtblFinals);
    cd = cd.baseClass;
}
```

Comment (WalterBright) [inline src/dclass.d]: yes, that looks better

**Comment by MartinNowak** [comment]
Auto-merge toggled on

**Pull request merged**